app/org/sagebionetworks/bridge/models/upload/Upload.java

```diff
@@ -1,18 +1,41 @@
 package org.sagebionetworks.bridge.models.upload;
 
+import java.util.List;
+
+import org.joda.time.LocalDate;
+
 /** Metadata for Bridge uploads. */
 public interface Upload {
+    /**
+     * If the upload is in a state where calling uploadComplete() and kicking off validation is a valid thing to do,
+     * then this method will return true. Otherwise, this method will return false.
+     */
+    boolean canBeValidated();
 
-    /** Upload ID. This is the key in the Dynamo DB table that uniquely identifies this upload. */
-    String getUploadId();
+    /** Name of the file to upload. */
+    String getFilename();
+
+    /** Health code of the user from which this upload originates from. */
+    String getHealthCode();
 
     /**
      * S3 object ID (key name). This is generated by Bridge to ensure no filename collisions and to ensure that we
      * don't get S3 hotspots from poorly distributed names.
      */
-    // TODO: This is being merged with Upload ID. Delete this field when the migration is complete.
     String getObjectId();
 
-    /** True if the upload is complete. False otherwise. */
-    boolean isComplete();
+    /** Represents upload status, such as requested, validation in progress, validation failed, or succeeded. */
+    UploadStatus getStatus();
+
+    /**
+     * Calendar date the file was uploaded (specifically, the uploadComplete() call. Date is determined using Pacific
+     * local time.
+     */
+    LocalDate getUploadDate();
+
+    /** Upload ID. This is the key in the Dynamo DB table that uniquely identifies this upload. */
+    String getUploadId();
+
+    /** List of validation messages, generally contains error messages. */
+    List<String> getValidationMessageList();
```

Comment:
Can you explain why we are expecting more than one message for a single upload?  An example would help.